opentracing-kotlin/src/main/kotlin/org/zalando/opentracing/kotlin/TracerExtension.kt

```diff
@@ -0,0 +1,90 @@
+package org.zalando.opentracing.kotlin
+
+import io.opentracing.Scope
+import io.opentracing.Span
+import io.opentracing.SpanContext
+import io.opentracing.Tracer
+import io.opentracing.propagation.Format
+import io.opentracing.propagation.TextMapAdapter
+import io.opentracing.tag.Tags
+
+inline fun <T : Tracer, R> T.trace(span: Span, block: (Span) -> R): R {
+
+    var scope: Scope? = null
+    var closed = false
+
+    try {
+        scope = scopeManager().activate(span)
+
+        return block(span)
+    } catch (e: Exception) {
+        span.captureException(e)
+        try {
+            scope?.close()
+            closed = true
+        } catch (closeException: Exception) {
+            // Do Nothing
+        }
+        throw e
+    } finally {
+        if (!closed)
+            scope?.close()
+        span.finish()
+    }
+}
+
+inline fun <T : Tracer, R> T.trace(
```

Comment:
I couldn't find a test for this, and means I couldn't see a usage either. With two function parameters, I'm expecting that the call site will look kinda weird.

Reply:
https://github.com/zalando/opentracing-toolbox/pull/472/files#diff-25419aff34533688bc24a3251cae001b66ba3d8995ccaca50d6fad5ae4af20c1R54

Reply:
2 function parameters is actually quite common in kotlin.

Reply:
usually, you should not require this method, but it is still there in case someone needs to do something different while creating the span. usually the other method with component and kind should suffice